app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,44 @@
+class MerchantsController < ApplicationController
+
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def item_index
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def new
+  end
+
+  def edit
+    @merchant_id = params[:id]
+  end
+
+  def create
+    Merchant.create!(merchant_params)
+    redirect_to '/merchants'
+  end
+
+  def delete
+    merchant = Merchant.find(params[:id])
+    merchant.items.destroy_all
+    merchant.destroy
+    redirect_to '/merchants'
+  end
+
+  def update
+    Merchant.update(merchant_params)
+    redirect_to "/merchants/#{params[:id]}"
+  end
```

Comment:
Same thing here as with the items update... if you prepopulate the edit form then this will allow you to update just parts of the merchant instead of needing to update everything in order for it to update.